src/app/users/users.component.ts

```diff
@@ -0,0 +1,122 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import User from '../User';
+import { UsersService } from '../users.service';
+
+@Component({
+	selector: 'app-users',
+	templateUrl: './users.component.html',
+	styleUrls: ['./users.component.css']
+})
+export class UsersComponent implements OnInit {
+
+	public users: User[] = [];
+
+	public openEditForm: boolean = false;
+
+	public userToEdit: User = new User();
+
+	public selectedUsers: User[] = [];
+
+	public isOpenConfirmDeleteSelectedUsersDialog: boolean = false;
+
+	public isOpenConfirmDeleteUserDialog: boolean = false;
+
+	public isFormOpen: boolean = false;
+
+	public columns: any[] = [];
+
+	public formType: "Add" | 'Edit' | '' =  '';
+
+	public userToDelete: User = new User();
+
+	public userForm: FormGroup = this.formBuilder.group({
+		firstName: ['', [Validators.required, Validators.minLength(4)]],
+		lastName: ['', [Validators.required, Validators.minLength(4)]],
+		email: ['', Validators.email],
+		username: ''
+	})
+
+	constructor(
+		private formBuilder: FormBuilder,
+		private userService: UsersService
+	) {	}
+
+	
+	ngOnInit(): void {
+		this.users = this.userService.users;		
+		this.columns = [
+			{ field: 'id', header: 'Id', width: '3%' },
+			{ field: 'firstName', header: 'First name', width: '16%' },
+			{ field: 'lastName', header: 'Last name', width: '16%' },
+			{ field: 'username', header: 'Username', width: '20%' },
+			{ field: 'email', header: 'Email', width: '23%' },
+			{ field: 'actions', header: 'Actions', width: '20%' },
+		];
+	}
+	
+	openForm(type: 'Add' | 'Edit', userId?: number){
+		this.formType = type;
+		this.isFormOpen = true;
+		if(type == 'Edit' && userId != undefined){
+			this.userToEdit = this.userService.findUserById(userId);
+		}
+		this.userForm.reset();
+	}
+	
+	openDeleteSelectedUsersDialog() {
+		this.isOpenConfirmDeleteSelectedUsersDialog = true;
+	}
+
+	openDeleteUserDialog(user: User){
+		this.userToDelete = user;
+		this.isOpenConfirmDeleteUserDialog = true;
+	}
+	
+	addUser() {
+		if(this.userForm.invalid){
+			return;
+		}
+		const user: User = this.fromFormToUser(this.userForm);
+		this.userService.addUser(user);		
+		this.userForm.reset();
+	}
+	
+	editUser(firstName: string, lastName: string, email: string) {
+		this.userService.editUser(this.userToEdit.id, firstName, lastName, email);
+	}
+		
+	deleteUser(user: User) {
+		this.userService.deleteUser(user);
+		this.isOpenConfirmDeleteUserDialog = false;
+	}
+	
+	deleteUsers() {
+		this.userService.deleteUsers(this.selectedUsers);
```

Comment:
send the selectedUserIds instead of all selected users
```suggestion
		this.userService.deleteUsers(this.selectedUsers.map((user: User) => user.id);
```